### PR TITLE
feature: Configure Apollo & Add Auth Flow (Sign up/in/out, Reset Password).

### PR DESCRIPTION
**Pull request by bobbytatum27**
### Administrative Info
Monday Board ID: 853606148
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?

- Added sign up, sign in, sign out, reset password
- Completed apollo setup

### Testing
How did you confirm your changes worked? 

- Manually went through each auth flow to confirm it worked as expected. See linked drive folder for demonstrations.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
[https://drive.google.com/drive/folders/1t2zWS9AtT5NlMqGCUV-qD0_qNP5j3Yiv?usp=sharing](url)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+Amplify.configure({
+  ...awsconfig,
+  Analytics: {
+    disabled: true,
+  },
+});
+
+// Initialize Apollo Client
+const client = new ApolloClient({
+  uri: 'localhost:4000/graphql',
```
Comment (c3duan) [inline src/App.tsx]: Can we load the path from an environment variable instead of hardcoding it here?

**Review by c3duan — COMMENTED** [review]
Overall everything looks great! 

Just a couple of small things to do:
- Delete `amplify/README.md` file
- Add `src/.eslintcache` to `.gitignore`
- Load graphql uri prefix from an environment variables

Some questions:
- Can you explain what the `testFunction` lambda function does for the sign-up and login flow? (Seems like it is not doing anything based on the implementation)
- From the demos, are all the passwords empty?

File: src/App.tsx (hunk 1 above)
Comment (bobbytatum27) [reply src/App.tsx:24]: > Overall everything looks great!
> 
> Just a couple of small things to do:
> 
> * Delete `amplify/README.md` file
> * Add `src/.eslintcache` to `.gitignore`
> * Load graphql uri prefix from an environment variables
> 
> Some questions:
> 
> * Can you explain what the `testFunction` lambda function does for the sign-up and login flow? (Seems like it is not doing anything based on the implementation)
> * From the demos, are all the passwords empty?

We'll get on the changes! As for the first question, it seems to be an auto-generated function that none of us caught. With regard to the demos, the passwords are not empty. It seems the ios screen record ignores text input for these fields as a built-in security measure. I tested this myself and can confirm that text is indeed entered but the screen capture doesn't record it.

**Comment by bobbytatum27** [comment]
> Overall everything looks great!
> 
> Just a couple of small things to do:
> 
> * Delete `amplify/README.md` file
> * Add `src/.eslintcache` to `.gitignore`
> * Load graphql uri prefix from an environment variables
> 
> Some questions:
> 
> * Can you explain what the `testFunction` lambda function does for the sign-up and login flow? (Seems like it is not doing anything based on the implementation)
> * From the demos, are all the passwords empty?



> Overall everything looks great!
> 
> Just a couple of small things to do:
> 
> * Delete `amplify/README.md` file
> * Add `src/.eslintcache` to `.gitignore`
> * Load graphql uri prefix from an environment variables
> 
> Some questions:
> 
> * Can you explain what the `testFunction` lambda function does for the sign-up and login flow? (Seems like it is not doing anything based on the implementation)
> * From the demos, are all the passwords empty?

We'll get on the changes! As for the first question, it seems to be an auto-generated function that none of us caught. I think it may have been created via an amplify mock. Since it doesn't do anything, should we remove it? 

With regard to the demos, the passwords are not empty. It seems the ios screen record ignores text input for these fields as a built-in security measure. I tested this myself and can confirm that text is indeed entered but the screen capture doesn't record it.

**Comment by sharmaadvay0** [comment]
A few quick questions so that I can start working on the onboarding page:

- Are we storing a [token from AWS](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html) in [secure storage](https://docs.expo.io/versions/latest/sdk/securestore/) so that we can authenticate backend requests and auto-login users after they have logged in once on their device?
- Are we setting a global state or context so that other pages can be conditionally rendered after someone has logged in or signed up?

Just as a side note, after this pull request is merged, I may have to add a function to the login page that queries the backend to check if the user has any home information.

**Comment by c3duan** [comment]
> A few quick questions so that I can start working on the onboarding page:
> 
> * Are we storing a [token from AWS](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html) in [secure storage](https://docs.expo.io/versions/latest/sdk/securestore/) so that we can authenticate backend requests and auto-login users after they have logged in once on their device?
> * Are we setting a global state or context so that other pages can be conditionally rendered after someone has logged in or signed up?
> 
> Just as a side note, after this pull request is merged, I may have to add a function to the login page that queries the backend to check if the user has any home information.

The Auth module from aws-amplify provides functionality to retrieve the currently logged-in user's credential. Please see this [example](https://github.com/aws-samples/aws-amplify-graphql/blob/b6fb8a5e16d955b4fd832442853181a3c0f1a8ef/src/App.js#L37).

**Comment by c3duan** [comment]
@bobbytatum27 , can we delete the `testFunction` if we don't need it?

**Comment by bobbytatum27** [comment]
> @bobbytatum27 , can we delete the `testFunction` if we don't need it?

Done!

File: src/App.tsx [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
+import React, { Fragment } from 'react';
+import { StyleSheet, SafeAreaView, View, AppRegistry } from 'react-native';
+import Amplify from 'aws-amplify';
+import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
+import awsconfig from './aws-exports';
+
+import AuthContainer from './components/auth-container';
+import SplashScreen from './components/splash-screen';
+
+const ApolloURI = 'localhost:4000/graphql';
```
Comment (c3duan) [inline src/App.tsx]: What I mean by an environmental variable is that you define the variable in an `.env` file, and you retrieve the variable ```const APOLLO_URI = `${process.env['APP_URI'] || 'localhost:4000'}/graphql`;```

File: src/components/splash-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { View, Image } from 'react-native';
+import SvgContainer from './SvgContainer';
+
+import SplashScreenStyles from '../styles/splash-screen-styles';
+
+// SVG xml must be converted to a string for react-native-svg
```
Comment (c3duan) [inline src/components/splash-screen.tsx]: One workaround is to make a separate react component with the svg which is a lot cleaner and we should store all the svg components in the an `assets` or `icons` folder

File: src/images/YCWLogo.svg [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+</defs>
+</svg>
```
Comment (c3duan) [inline src/images/YCWLogo.svg]: Make this svg into a react functional component.

**Review by c3duan — CHANGES_REQUESTED** [review]
@bobbytatum27 @hanakims @TheHiddenLayer A few more changes need to be addressed before merging.

**Comment by sharmaadvay0** [comment]
> > A few quick questions so that I can start working on the onboarding page:
> > 
> > * Are we storing a [token from AWS](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html) in [secure storage](https://docs.expo.io/versions/latest/sdk/securestore/) so that we can authenticate backend requests and auto-login users after they have logged in once on their device?
> > * Are we setting a global state or context so that other pages can be conditionally rendered after someone has logged in or signed up?
> > 
> > Just as a side note, after this pull request is merged, I may have to add a function to the login page that queries the backend to check if the user has any home information.
> 
> The Auth module from aws-amplify provides functionality to retrieve the currently logged-in user's credential. Please see this [example](https://github.com/aws-samples/aws-amplify-graphql/blob/b6fb8a5e16d955b4fd832442853181a3c0f1a8ef/src/App.js#L37).

@c3duan Will these features be added to this PR?

**Comment by c3duan** [comment]
> > > A few quick questions so that I can start working on the onboarding page:
> 
> > > 
> 
> > > * Are we storing a [token from AWS](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html) in [secure storage](https://docs.expo.io/versions/latest/sdk/securestore/) so that we can authenticate backend requests and auto-login users after they have logged in once on their device?
> 
> > > * Are we setting a global state or context so that other pages can be conditionally rendered after someone has logged in or signed up?
> 
> > > 
> 
> > > Just as a side note, after this pull request is merged, I may have to add a function to the login page that queries the backend to check if the user has any home information.
> 
> > 
> 
> > The Auth module from aws-amplify provides functionality to retrieve the currently logged-in user's credential. Please see this [example](https://github.com/aws-samples/aws-amplify-graphql/blob/b6fb8a5e16d955b4fd832442853181a3c0f1a8ef/src/App.js#L37).
> 
> 
> 
> @c3duan Will these features be added to this PR? 

Let's add these in the next PR.

File: src/App.tsx (hunk 2 above)
Comment (bobbytatum27) [reply src/App.tsx]: We're working on this one. Based on the [Expo docs](https://docs.expo.io/guides/environment-variables/), it seems we need to do some extra work to access environment variables. [This article](https://alxmrtnz.com/thoughts/2019/03/12/environment-variables-and-workflow-in-expo.html) and [this one too](https://medium.com/swlh/how-to-properly-use-environment-variables-in-an-expo-react-native-app-7ab852590b30) propose solutions to access environment variables in Expo.

**Review by c3duan — APPROVED** [review]
@bobbytatum27 @hanakims @TheHiddenLayer Good job on addressing my requests in your first PR. LGTM! Remember to squash and merge to clean up the commit message.

**Review by sharmaadvay0 — APPROVED** [review]
Looks good. I'll start working on integrating these changes with the onboarding page.

**Pull request merged**